website/addons/evernote/views.py

```diff
@@ -0,0 +1,215 @@
+    must_be_contributor_or_public,
+    must_have_addon,
+    must_be_addon_authorizer,
+    must_not_be_registration,
+    must_have_permission
+)
+
+import ENML2HTML
+
+import logging
+logger = logging.getLogger(__name__)
+
+SHORT_NAME = 'evernote'
+FULL_NAME = 'Evernote'
+
+@must_be_logged_in
+def evernote_get_user_settings(auth):
+    """ Returns the list of all of the current user's authorized Evernote accounts """
+    serializer = EvernoteSerializer(user_settings=auth.user.get_addon('evernote'))
+    return serializer.serialized_user_settings
+
+
+@must_have_addon('evernote', 'node')
+@must_have_permission(permissions.WRITE)
+def evernote_get_config(node_addon, auth, **kwargs):
+    """API that returns the serialized node settings."""
+    # following from box addon:
+    # if node_addon.external_account:
+    #     refresh_oauth_key(node_addon.external_account)
+
+    #logger.debug(node_addon)
+    return {
+        'result': EvernoteSerializer().serialize_settings(node_addon, auth.user),
+    }
+
+@must_not_be_registration
+@must_have_addon('evernote', 'user')
+@must_have_addon('evernote', 'node')
+@must_be_addon_authorizer('evernote')
+@must_have_permission(permissions.WRITE)
+def evernote_set_config(node_addon, user_addon, auth, **kwargs):
+    """View for changing a node's linked Evernote folder."""
+    folder = request.json.get('selected')
+    serializer = EvernoteSerializer(node_settings=node_addon)
+
+    uid = folder['id']
+    path = folder['path']
+
+    node_addon.set_folder(uid, auth=auth)
+
+    return {
+        'result': {
+            'folder': {
+                'name': path.replace('All Files', '') if path != 'All Files' else '/ (Full Evernote)',
+                'path': path,
+            },
+            'urls': serializer.addon_serialized_urls,
+        },
+        'message': 'Successfully updated settings.',
+    }
+
+
+@must_have_addon('evernote', 'user')
+@must_have_addon('evernote', 'node')
+@must_have_permission(permissions.WRITE)
+def evernote_add_user_auth(auth, node_addon, user_addon, **kwargs):
+    """Import evernote credentials from the currently logged-in user to a node.
+    """
+    external_account = ExternalAccount.load(
+        request.json['external_account_id']
+    )
+
+    if external_account not in user_addon.external_accounts:
+        raise HTTPError(http.FORBIDDEN)
+
+    try:
+        node_addon.set_auth(external_account, user_addon.owner)
+    except PermissionsError:
+        raise HTTPError(http.FORBIDDEN)
+
+    node_addon.set_user_auth(user_addon)
+    node_addon.save()
+
+    return {
+        'result': EvernoteSerializer().serialize_settings(node_addon, auth.user),
+        'message': 'Successfully imported access token from profile.',
+    }
+
+
+evernote_deauthorize_node = generic_views.deauthorize_node(
+    SHORT_NAME
+)
+
+@must_have_addon('evernote', 'node')
+@must_be_addon_authorizer('evernote')
+def evernote_notes(node_addon, **kwargs):
```

Comment:
This functionality will likely be moved into WaterButler -- it appears to be retrieving metadata about notes, preparing them for display.